src/libs/workspace-utils.ts

```diff
@@ -71,6 +71,15 @@ export interface WorkspaceSubmissionStats {
   runningSubmissionsCount: number;
 }
 
+export interface WorkspaceAccessEntry {
+  accessLevel: WorkspaceAccessLevel;
+  pending: boolean;
+  canShare: boolean;
+  canCompute: boolean;
+}
+
+export type WorkspaceACL = Record<string, WorkspaceAccessEntry>;
```

Comment:
Not sure what the keys in this object contain, but the mapped object form of this type might be a little better for documentation.

For example, if the keys contain user emails:
```suggestion
export type WorkspaceACL = { [userEmail: string]: WorkspaceAccessEntry };
```